pyvista/core/pointset.py

```diff
@@ -142,50 +197,74 @@ class PolyData(_vtk.vtkPolyData, PointSet, PolyDataFilters):
-                self._from_file(args[0])
-            elif isinstance(args[0], (np.ndarray, list)):
-                if isinstance(args[0], list):
-                    points = np.asarray(args[0])
-                    if not np.issubdtype(points.dtype, np.number):
-                        raise TypeError('Points must be a numeric type')
-                else:
-                    points = args[0]
-                if points.ndim != 2:
-                    points = points.reshape((-1, 3))
-                cells = self._make_vertex_cells(points.shape[0])
-                self._from_arrays(points, cells, deep, verts=True)
-            else:
-                raise TypeError('Invalid input type')
 
-        elif len(args) == 2:
-            arg0_is_array = isinstance(args[0], np.ndarray)
-            arg1_is_array = isinstance(args[1], np.ndarray)
-            if arg0_is_array and arg1_is_array:
-                self._from_arrays(args[0], args[1], deep)
+        # PolyData-like
+        elif isinstance(var_inp, (PolyData, _vtk.vtkPolyData)):
+            for kwarg in ['faces', 'n_faces', 'lines']:
+                if local_parms[kwarg]:
+                    raise ValueError('No other arguments should be set when first '
+                                     'parameter is a PolyData')
+            if deep:
+                self.deep_copy(var_inp)
             else:
-                raise TypeError('Invalid input type')
-        else:
-            raise TypeError('Invalid input type')
+                self.shallow_copy(var_inp)
+            return
 
-        # Check if need to make vertex cells
-        if self.n_points > 0 and self.n_cells == 0:
-            # make vertex cells
-            self.verts = self._make_vertex_cells(self.n_points)
+        # First parameter is points
+        elif isinstance(var_inp, (np.ndarray, list)):
+            points = var_inp
+            if not isinstance(points, np.ndarray):
+                points = np.array(points)
+            self.SetPoints(pyvista.vtk_points(points, deep=deep))
+        else:
+            TypeError('Invalid Input type:\n\n'
+                      'Expected first argument to be either a:\n'
+                      '    - vtk.PolyData\n'
+                      '    - pyvista.PolyData\n'
+                      '    - numeric numpy.ndarray (1 or 2 dimensions)\n'
+                      '    - List (flat or nested with 3 points per vertex)\n\n'
+                      f'Instead got: {type(var_inp)}')
+
+        # At this point, points have been setup, add faces and/or lines
+        if faces is None and lines is None:
+            # one cell per point (point cloud case)
+            verts = self._make_vertex_cells(self.n_points)
+            self.verts = CellArray(verts, self.n_points, deep)
+
+        elif faces is not None:
+            # here we use CellArray since we must specify deep and n_faces
+            self.faces = CellArray(faces, n_faces, deep)
```

Comment:
Is it expected that we _either_ set `.verts` _or_ `.faces`? I'm not familiar enough with PolyData to be sure, just checking. Hmm, looking at the helper function you tore out it's probably expected. I'll leave this here anyway to be on the safe side.

Reply:
I like "tore out".  Helper function wasn't necessary as the main logic is only used here and this really simplifies it.

I was confused about the implementation as well, and we can actually have lines, vertices, and faces.  The idea is that you can plot all three at the same time.  When creating the mesh, however, we only want to use vertices in the "point cloud" case, where each point is given a single point cell.